lib/schemas.js

```diff
@@ -86,7 +86,22 @@ function normalizeSchema (routeSchemas, serverOptions) {
   if (routeSchemas.response) {
     const httpCodes = Object.keys(routeSchemas.response)
     for (const code of httpCodes) {
-      routeSchemas.response[code] = getSchemaAnyway(routeSchemas.response[code], serverOptions.jsonShorthand)
+      const contentProperty = routeSchemas.response[code].content
+
+      // ducktyping to check if the media content is meant to be used or it's the short form.
+      let mediaContent = false
+      if (contentProperty) {
+        const firstEntry = contentProperty[Object.keys(contentProperty)[0]]
+        mediaContent = !!(typeof firstEntry === 'object' && firstEntry.schema)
```

Comment:
What does the `object.keys(contentProperty).length > 1` indicates? 

The purpose of this improvement is to make sure that all keys have `scheme` if multiple content types is meant to be used. right?

throwing an error if `index !==0` means that some of the keys had `scheme` and others not, so maybe we should throw something similar to `schema is missing for the ${property}`. 

but what about the object.keys check?

Reply:
The `Object.keys(contentProperty)` is run twice at line 94 and 99 (`for (const mediaName in contentProperty)`)

> What does the object.keys(contentProperty).length > 1 indicates?

Instead of creating the `mediaContent` boolean, you could write (metacode):

```
if(contentProperty) {
  const keys = Object.keys(contentProperty)
  for (const i = 0; i < keys.length; i++) {
    const mediaName = keys[i]
    if(!contentProperty[mediaName].schema && keys.length>1){
      throw must have schema
    }
    routeSchemas.response[code].content[mediaName].schema = getSchemaAnyway(contentProperty[mediaName].schema, serverOptions.jsonShorthand)
  }
```

so you don't have a 2 step check and it is more consistent and isolated

Reply:
Perfect, thank you for the elaboration, I've committed the changes, let me know if anything else could also be improved